### PR TITLE
Adding agent version in RM CI events. 

### PR DESCRIPTION
**Pull request by kasubram**
Marking artifact cleanup failure as user error.

### CODE REVIEW

File: src/Agent.Worker/Release/ReleaseJobExtension.cs [hunk 1]
```diff
@@ -207,7 +207,14 @@ public ReleaseJobExtension()
                         {
                             var releaseFileSystemManager = HostContext.GetService<IReleaseFileSystemManager>();
                             executionContext.Output(StringUtil.Loc("RMEnsureArtifactFolderExistsAndIsClean", downloadFolderPath));
-                            releaseFileSystemManager.CleanupDirectory(downloadFolderPath, executionContext.CancellationToken);
+                            try
+                            {
+                                releaseFileSystemManager.CleanupDirectory(downloadFolderPath, executionContext.CancellationToken);
+                            }
+                            catch(Exception ex) when (ex is DirectoryNotFoundException || ex is UnauthorizedAccessException)
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: space after catch

File: src/Agent.Worker/Release/ReleaseJobExtension.cs [hunk 2]
```diff
@@ -207,7 +207,14 @@ public ReleaseJobExtension()
...
+                            {
+                                throw new ArtifactCleanupFailedException(StringUtil.Loc("FailedCleaningupRMArtifactDirecotyr", downloadFolderPath), ex);
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: typo: FailedCleaningupRMArtifactDirecotyr

File: src/Agent.Worker/Release/ReleaseJobExtension.cs (hunk 1 above)
Comment (kasubram) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: done

File: src/Agent.Worker/Release/ReleaseJobExtension.cs (hunk 2 above)
Comment (kasubram) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: done.

File: src/Misc/layoutbin/en-US/strings.json [hunk 3]
```diff
@@ -76,6 +76,7 @@
   "ErrorOccurredWhilePublishingCCFiles": "Error occurred while publishing code coverage files. Error: {0}",
   "EulasSectionHeader": "End User License Agreements",
   "Failed": "Failed: ",
+  "FailedCleaningupRMArtifactDirectory": "Failed to cleanup RM artifact directory '{0}'.",
```
Comment (GitHubSriramB) [inline src/Misc/layoutbin/en-US/strings.json]: .. cleanup Release's artifacts directory...

File: src/Agent.Worker/Release/ReleaseJobExtension.cs [hunk 4]
```diff
@@ -207,7 +207,14 @@ public ReleaseJobExtension()
                         {
                             var releaseFileSystemManager = HostContext.GetService<IReleaseFileSystemManager>();
                             executionContext.Output(StringUtil.Loc("RMEnsureArtifactFolderExistsAndIsClean", downloadFolderPath));
-                            releaseFileSystemManager.CleanupDirectory(downloadFolderPath, executionContext.CancellationToken);
+                            try
+                            {
+                                releaseFileSystemManager.CleanupDirectory(downloadFolderPath, executionContext.CancellationToken);
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: good to add retry also?

File: src/Agent.Worker/Release/ReleaseJobExtension.cs [hunk 5]
```diff
@@ -221,25 +228,37 @@ public ReleaseJobExtension()
...
-                    IOUtil.DeleteDirectory(artifactsWorkingFolder, executionContext.CancellationToken);
-                }
 
-                Directory.CreateDirectory(artifactsWorkingFolder);
-            }
-            catch (Exception ex)
+            RetryExecutor retryExecutor = new RetryExecutor();
+            retryExecutor.ShouldRetryAction = (ex) =>
             {
+                executionContext.Output(StringUtil.Loc("FailedCleaningupRMArtifactDirectory", artifactsWorkingFolder));
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: Good to Output the exception also (assuming Trace.Error will not log it to execution output)?
Also good to log that we are retrying the cleanup? Essentially - you can do:
Retrying the cleanup of Release artifacts directory, exception: {0}

**Pull request merged**